### PR TITLE
Agent - allow the user to set the agent address - ManagerAddress

### PR DESCRIPTION
**Pull request by ygalblum**
Add a configuration flag in both the config file and command line arguments for setting the hirte bus address If the address is set via config or arguments use it instead of host and port

Add the configuration flag to the example config file Update the man page for the executable to conf file

### CODE REVIEW

File: doc/man/hirte-agent.1.md [hunk 1]
```diff
@@ -30,6 +30,12 @@ The host used by `hirte-agent` to connect to `hirte`. Must be a valid IPv4 or IP
 
 The port on which `hirte` is listening for connection request and the `hirte-agent` is connecting to. This option will overwrite the port defined in the configuration file.
 
+#### **--address**, **-a**
+
+SD Bus address used by `hirte-agent` to connect to `hirte`. See `man sd_bus_set_address` for its format.
```
Comment (alexlarsson) [inline doc/man/hirte-agent.1.md]: Just call it "dbus address". This is a standard format used by all dbus libraries, specified in the dbus spec.

Comment (alexlarsson) [reply doc/man/hirte-agent.1.md]: reference: 
https://dbus.freedesktop.org/doc/dbus-specification.html#addresses

**Comment by alexlarsson** [comment]
Other than the docs comment, and the code style issue, this looks good to me.

File: doc/man/hirte-agent.1.md (hunk 1 above)
Comment (ygalblum) [reply doc/man/hirte-agent.1.md]: Fixed

File: src/agent/agent.c [hunk 2]
```diff
@@ -370,27 +370,27 @@ bool agent_set_port(Agent *agent, const char *port_s) {
         return true;
 }
 
-bool agent_set_host(Agent *agent, const char *host) {
-        char *dup = strdup(host);
+static bool agent_set_str(char **p, const char *s) {
```
Comment (engelmi) [inline src/agent/agent.c]: Since this is just copying the source to the target, how about we move it to `libhirte/common/common.c` (needs to be created) and rename it to `copy_string` or sth similar? This could simplify quite some code throughout hirte. 
Edit: Refactoring of the other parts is not part of this PR, of course.

File: doc/man/hirte-agent.1.md [hunk 3]
```diff
@@ -30,6 +30,12 @@ The host used by `hirte-agent` to connect to `hirte`. Must be a valid IPv4 or IP
 
 The port on which `hirte` is listening for connection request and the `hirte-agent` is connecting to. This option will overwrite the port defined in the configuration file.
 
+#### **--address**, **-a**
+
+DBus address used by `hirte-agent` to connect to `hirte`. See `man sd_bus_set_address` for its format.
+Overrides any setting of `ManagerHost` or `ManagerPort`.
+This option will overwrite the port defined in the configuration file.
```
Comment (engelmi) [inline doc/man/hirte-agent.1.md:37]:
```suggestion
This option will overwrite the host and port defined in the configuration file.
```

Comment (engelmi) [inline doc/man/hirte-agent.1.md]:
```suggestion
Overrides any setting of `ManagerHost` or `ManagerPort` defined in the configuration file as well as the respective CLI options. 
```

Comment (ygalblum) [reply doc/man/hirte-agent.1.md]: Done. I've changed it in both files (since the address takes precedence over any host:port configuration)

File: src/agent/agent.c (hunk 2 above)
Comment (ygalblum) [reply src/agent/agent.c]: Done

**Review by engelmi — APPROVED** [review]
LGTM

**Comment by ygalblum** [comment]
Resolves #206
